thentos-adhocracy/src/Thentos/Adhocracy3/Types.hs

```diff
@@ -127,28 +147,30 @@ instance ThentosErrorToServantErr DB where
         -- For errors specifically relevant to the A3 frontend we mirror the A3 backend errors
         -- exactly so that the frontend recognizes them
         f (ThentosA3ErrorCore e) = case e of
-            BadCredentials -> (Nothing, mkA3StyleServantErr err400 $ A3ErrorInfo
+            BadCredentials -> (Nothing, mkA3StyleServantErr err400 $ A3Error
                 "password"
                 "body"
                 "User doesn't exist or password is wrong")
-            UserEmailAlreadyExists -> (Nothing, mkA3StyleServantErr err400 $ A3ErrorInfo
+            UserEmailAlreadyExists -> (Nothing, mkA3StyleServantErr err400 $ A3Error
                 "data.adhocracy_core.sheets.principal.IUserExtended.email"
                 "body"
                 "The user login email is not unique")
-            UserNameAlreadyExists -> (Nothing, mkA3StyleServantErr err400 $ A3ErrorInfo
+            UserNameAlreadyExists -> (Nothing, mkA3StyleServantErr err400 $ A3Error
                 "data.adhocracy_core.sheets.principal.IUserBasic.name"
                 "body"
                 "The user login name is not unique")
-            NoSuchPendingUserConfirmation -> (Nothing, mkA3StyleServantErr err400 $ A3ErrorInfo
+            NoSuchPendingUserConfirmation -> (Nothing, mkA3StyleServantErr err400 $ A3Error
                 "path"
                 "body"
                 "Unknown or expired activation path")
-            NoSuchThentosSession -> (Nothing, mkA3StyleServantErr err400 $ A3ErrorInfo
+            NoSuchThentosSession -> (Nothing, mkA3StyleServantErr err400 $ A3Error
                 "X-User-Token"
                 "header"
                 "Invalid user token")
             _ -> thentosErrorToServantErr (Just mkErr) e
 
+        f (GenericA3Error errMsg) =
+            (Nothing, err400 {errBody = encode errMsg, errHeaders = [contentTypeJsonHeader]} )
```

Comment:
It seems like `400` can be misleading here. E.g. if A3 is down we probably want 502. I know you don't want to fully parse the possible errors, but would relaying the status code from A3 suffice?